Support "bpo-" in Misc/NEWS

brettcannon opened the pull request.

brettcannon committed:
Support "bpo-" in Misc/NEWS

Carreau commented on Doc/tools/extensions/pyspecific.py:
```diff
@@ -253,7 +253,7 @@ def run(self):
             node = nodes.strong(text, text)
             return [node]
-        content = issue_re.sub(r'`\1ssue #\2 <https://bugs.python.org/\2>`__',
+        content = issue_re.sub(r'`\bpo-\2 <https://bugs.python.org/\2>`__',
```
`\b` ? I think you overwrote the `\1` :-)

brettcannon replied:
That's on purpose as the first capture group is the prefix which doesn't matter. Probably should change the regex to use a non-capture group.

Carreau replied:
Sure, you can either you use `\1` or insert `bpo-`. But here you overwrote half the capture group by only deleting the `1` and not the `\`.

brettcannon replied:
Ah, you're right. Now fixed.

brettcannon committed:
Fix up the match and substitution regexes

Make the prefix match non-capturing and fix a bug where a backslash was left into the substitution.

zware commented on Doc/tools/extensions/pyspecific.py:
```diff
@@ -34,7 +34,7 @@
 
 ISSUE_URI = 'https://bugs.python.org/issue%s'
-SOURCE_URI = 'https://hg.python.org/cpython/file/3.6/%s'
+SOURCE_URI = 'https://github.com/python/cpython/tree/3.6/%s'
```
Shouldn't it be `master` rather than `3.6` on this branch?

Mariatta replied:
My patch for another issue will address this, I think 🤔 

http://bugs.python.org/file45904/issue28941.patch

brettcannon replied:
I went ahead and updated here since I'm already changing it.

Mariatta replied:
ok :)

zware commented on Doc/tools/extensions/pyspecific.py:
```diff
@@ -253,7 +253,7 @@ def run(self):
             node = nodes.strong(text, text)
             return [node]
-        content = issue_re.sub(r'`\1ssue #\2 <https://bugs.python.org/\2>`__',
+        content = issue_re.sub(r'`bpo-\1 <https://bugs.python.org/\1>`__',
```
I'd suggest making it `https://bugs.python.org/issue\1` to avoid the redirect through https://www.python.org/sf/

brettcannon replied:
Done.

brettcannon committed:
Address reviewer comments

Clucier17 reviewed (commented):
<Spam>
